OpenSim/Simulation/Model/PointForceDirection.h

```diff
@@ -23,70 +23,74 @@
-/** Convenience class for a generic representation of geometry of a complex
-    Force (or any other object) with multiple points of contact through
-    which forces are applied to bodies. This represents one such point and an
-    array of these objects defines a complete Force distribution (ie. path).
+
+/**
+ * Convenience class for a generic representation of geometry of a complex
+ * Force (or any other object) with multiple points of contact through
+ * which forces are applied to bodies. This represents one such point and an
+ * array of these objects defines a complete Force distribution (ie. path).
  *
  * @author Ajay Seth
  * @version 1.0
  */
+class OSIMSIMULATION_API PointForceDirection final {
+public:
+    PointForceDirection(
+        SimTK::Vec3 point,
+        const PhysicalFrame& frame,
+        SimTK::Vec3 direction) :
 
-class OSIMSIMULATION_API PointForceDirection
-{
+        _point(point), _frame(&frame), _direction(direction)
+    {}
 
-//=============================================================================
-// MEMBER VARIABLES
-//=============================================================================
-private:
-    /** Point of "contact" with a body, defined in the body frame */
-    SimTK::Vec3 _point;
-    /** The frame in which the point is defined */
-    const PhysicalFrame &_frame;
-    /** Direction of the force at the point, defined in ground */
-    SimTK::Vec3 _direction;
-    /** Optional parameter to scale the force that results from a scalar 
-        (tension) multiplies the direction */
-    double _scale;
-//=============================================================================
-// METHODS
-//=============================================================================
-    //--------------------------------------------------------------------------
-    // CONSTRUCTION
-    //--------------------------------------------------------------------------
-public:
-    virtual ~PointForceDirection() {};
-    /** Default constructor takes the point, body, direction and scale
-        as arguments */
-    PointForceDirection(SimTK::Vec3 point, const PhysicalFrame &frame, 
-        SimTK::Vec3 direction, double scale=1):
-            _point(point), _frame(frame), _direction(direction), _scale(scale)
+    [[deprecated("the 'scale' functionality should not be used in new code: OpenSim already assumes 'direction' is non-unit-length")]]
+    PointForceDirection(
+        SimTK::Vec3 point,
+        const PhysicalFrame& frame,
+        SimTK::Vec3 direction,
+        double scale) :
+
+        _point(point), _frame(&frame), _direction(direction), _scale(scale)
     {}
 
     /** get point of "contact" with on a body defined in the body frame */
-    SimTK::Vec3 point() {return _point; }
+    SimTK::Vec3 point() { return _point; }
+
     /** get the body in which the point is defined */
-    const PhysicalFrame& frame() {return _frame; }
+    const PhysicalFrame& frame() { return *_frame; }
```

Comment:
Just to bring this up to date, I think the comment on L66 (and similar) should refer to `PhysicalFrame` rather than "body" since not all `PhysicalFrame`s are `Body`s.